### PR TITLE
Nodemailer adapter

### PR DESCRIPTION
**Pull request by stefan-prokop-cz**
* This PR adds custom adapter which use `nodemailer` to send emails
* Updates docs
* Adds tests

### CODE REVIEW

File: src/lib/adapters/CustomAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import * as nodemailer from 'nodemailer';
+import { Options } from 'nodemailer/lib/smtp-transport';
+import { Adapter, AdapterSend } from '../enmail';
+
+export type CustomOptions = Options;
+
+export const createCustom = (adapterOptions: CustomOptions): Adapter<CustomOptions, Partial<nodemailer.SendMailOptions>> => {
```
Comment (stefan-prokop-cz) [inline src/lib/adapters/CustomAdapter.ts:7]: I am thinking about something like nodemailer adapter

File: src/lib/adapters/SendgridAdapter.ts [hunk 2]
```diff
@@ -13,7 +14,7 @@ export const createSendgrid = (adapterOptions: SendgridOptions): Adapter<Sendgri
         options: adapterOptions,
         mailService: sendgrid,
         sender: (sendOptions) => {
-            const send: AdapterSend<Partial<MailData>> = (message, options) => {
+            const send: AdapterSend<Partial<SendgridMessage>> = (message, options) => {
```
Comment (grissius) [inline src/lib/adapters/SendgridAdapter.ts:17]: NIT: This change probably slipped by and is likely not related to the PR. No need to rebase tho.

File: src/test/integration/nodemailer.test.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        expect(adapter.options).toStrictEqual(options);
+        expect(typeof adapter.mailService.sendMail).toEqual('function');
+        expect(adapter).toMatchSnapshot();
+    });
+    it('Sender will be successfully returned', () => {
+        const sender = enmail.sender(adapter);
+        expect(typeof sender).toEqual('function');
+        expect(sender).toEqual(adapter.sender);
+    });
+    it('An email can be sent via adapter', async () => {
```
Comment (grissius) [inline src/test/integration/nodemailer.test.ts:36]: This test name is grossly misleading. We are just testing that an email object was created, not actually sending anything.

Comment (grissius) [reply src/test/integration/nodemailer.test.ts:36]: Same for all the remaining tests

File: src/test/integration/nodemailer.test.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        const gmailMessage = await adapter.sender()(body, sendOptions) as any;
+        expect(Object.keys(gmailMessage).sort()).toEqual(['bcc', 'to', 'html', 'from', 'subject'].sort());
```
Comment (grissius) [inline src/test/integration/nodemailer.test.ts:38]: This expect is repeated 7 times without a change. Should be refactored into a function.

File: src/test/integration/nodemailer.test.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+jest.mock('nodemailer');
+const nodemailer = require('nodemailer');
+nodemailer.createTransport.mockReturnValue({ sendMail: jest.fn(data => data) });
+
+describe('Nodemailer adapter', () => {
+    const getGmailAdapter = () => enmail.createAdapter.nodemailer(options);
+    const options = {
+        service: 'gmail',
+        auth: {
+            type: 'login' as any,
```
Comment (grissius) [inline src/test/integration/nodemailer.test.ts:12]: Is the cast required? If so, I think we ought to fix that. Devs that want to use the library will peek into tests, if they don't find what they're looking for in the readme, so they'll just copy that.

**Review by grissius — CHANGES_REQUESTED** [review]
There are some minor change suggestions in the tests. Feel free to discuss. :cat:

File: src/lib/adapters/SendgridAdapter.ts (hunk 2 above)
Comment (stefan-prokop-cz) [reply src/lib/adapters/SendgridAdapter.ts:17]: yes i know, I needed to add this message type to the nodemailer so i did the same thing here without creating the new PR

File: src/test/integration/nodemailer.test.ts (hunk 3 above)
Comment (stefan-prokop-cz) [reply src/test/integration/nodemailer.test.ts]: I changed the description

File: src/test/integration/nodemailer.test.ts (hunk 4 above)
Comment (stefan-prokop-cz) [reply src/test/integration/nodemailer.test.ts]: i created `expect` function

File: src/test/integration/nodemailer.test.ts (hunk 5 above)
Comment (stefan-prokop-cz) [reply src/test/integration/nodemailer.test.ts]: i changed the `as any` to `as const` it looks there is no enum which we can use

**Review by grissius — APPROVED** [review]

**Review by smoliji — APPROVED** [review]

**Pull request merged**